### ISSUE TITLE
Alert provisioning does not support using Environment Variables

### ISSUE THREAD

**Issue by JamesGibo** [grafana/grafana#54984]
**What happened**:

When provisioning a slack alerting contact point, using an environment variable to set the URL, the provisioning template does not expand the environment variable. The same process works fine for other config files like when setting the URL for a Datasource.

https://grafana.com/docs/grafana/latest/administration/provisioning/#contact-points
https://grafana.com/docs/grafana/latest/administration/provisioning/#using-environment-variables

![image](https://user-images.githubusercontent.com/22477854/189369407-57f8ee62-8110-4c92-b7bf-0006cc15c80d.png)

**What you expected to happen**:

The provisioning template should expand the environment variable in alerting provisioning.

**How to reproduce it (as minimally and precisely as possible)**:
Use the following configuration  file and reload grafana
```yaml
# config file version
apiVersion: 1

contactPoints:
  - orgId: 1
    name: slack_1
    receivers:
      - uid: slack_1
        type: slack
        settings:
          # <string, required>
          recipient: alerting-dev
          url: ${SLACK_WEBHOOK}
          title: |
            {{ template "slack.default.title" . }}
          text: |
            {{ template "slack.default.text" . }}
```

**Anything else we need to know?**:

**Environment**:
- Grafana version: v9.1.2
- Data source type & version: 
- OS Grafana is installed on: Docker
- User OS & Browser: Chrome, Ubuntu 20.04
- Grafana plugins:
- Others:

**Comment by zuchka** [comment]
Thanks for creating this issue, @JamesGibo 

this sounds similar to the issue fixed by this PR. Perhaps helpful?

https://github.com/grafana/grafana/pull/54293

**Comment by bygui86** [comment]
Same issue here with Grafana v9.1.4
Here my screenshot: 
![Screenshot 2022-09-13 at 10 45 00](https://user-images.githubusercontent.com/22518997/189855824-0de09932-8c4d-4d5d-ae5f-b6e432a04fff.png)

@zuchka sorry but I don't think this issue is related to the one you mentioned.

@JamesGibo I have issues loading notification policies through provisioning, do you?
I'm not sure if alerting provisioning is 100% ready, I see tons of issues related to that in the last days.

**Comment by romankabakov** [comment]
I have the same issue with PagerDuty integration:
`               integrationKey: ${integrationKey}
`
Throw me an error
![Screenshot 2022-09-13 at 12 16 13](https://user-images.githubusercontent.com/28570779/189863225-0d0a7d22-1f70-4411-9948-d492270d654e.png)

I did check the environment variable exists and is set correctly.

**Comment by JamesGibo** [comment]
> I have issues loading notification policies through provisioning, do you?

I have managed to load notification polices OK through provisioning, but the alert provisioning documentation is lacking some key information and clear examples, I have had to look at the code a number of times.

**Comment by JamesGibo** [comment]
> Thanks for creating this issue, @JamesGibo
> 
> this sounds similar to the issue fixed by this PR. Perhaps helpful?
> 
> #54293

@zuchka I don't think this issue is related, this PR bypasses interpolation for query data for alerting rules, where I have seen the issue is configuring the contact point not the rules, these are 2 separate files.

Interestingly the alert rules file does interpolate other fields meaning I had to escape the `$`
```yaml
        # <duration, required> for how long should the alert fire before alerting
        for: 60s
        # <map<string, string>> a map of strings to pass around any data
        annotations:
          summary: "High CPU usage {{ $$labels.host }}"
          description: "Average CPU usage for host {{ $$labels.host }} is above 60%"
        # <map<string, string> a map of strings that can be used to filter and
        #                      route alerts
        labels:
          severity: Warning
```

**Comment by bygui86** [comment]
@JamesGibo can you please paste here an example of notification policies yaml file?
I'm struggling a lot and the documentation lacks quite important information.
I'm able to load only 1 notification policy as default, the last listed in yaml file, all others are kind of ignored or overwriting each other...

**Comment by JamesGibo** [comment]
> @JamesGibo can you please paste here an example of notification policies yaml file? I'm struggling a lot and the documentation lacks quite important information. I'm able to load only 1 notification policy as default, the last listed in yaml file, all others are kind of ignored or overwriting each other...

I only have a single notification policy so have not tested multiple I am afraid.

**Comment by bygui86** [comment]
@JamesGibo ok no worries, but I have also another issue. Once the single notification policy is provisioned, then I'm not able to create other policies in Grafana UI. This sounds really weird... or am I missing something?
Here a screenshot
![Screenshot 2022-09-13 at 10 57 56](https://user-images.githubusercontent.com/22518997/189871578-e25efdd8-51b3-401f-ad38-4c92d530b118.png)
Under section "Specific routing" I should be able to create further policies right?
What about your experience?

**Comment by JamesGibo** [comment]
> @JamesGibo ok no worries, but I have also another issue. Once the single notification policy is provisioned, then I'm not able to create other policies in Grafana UI. This sounds really weird... or am I missing something? Here a screenshot ![Screenshot 2022-09-13 at 10 57 56](https://user-images.githubusercontent.com/22518997/189871578-e25efdd8-51b3-401f-ad38-4c92d530b118.png) Under section "Specific routing" I should be able to create further policies right? What about your experience?

I would create a new issue for this as it is not related to the current issue, but i see the same.

**Comment by bygui86** [comment]
@JamesGibo just wanted to check with you.
I created a new issue right now: #55073

**Comment by JamesGibo** [comment]
Issue is still present in Grafana v9.1.5

**Comment by bygui86** [comment]
As alerting system is made of different components (contact points, templates, notification policies and alert rules) it would be really nice and important to check that each of them supports environment variables during provisioning.

**Comment by JamesGibo** [comment]
Issue is still present in Grafana v9.1.6

**Comment by armandgrillet** [comment]
Thanks for creating this issue, an engineer at Grafana Labs will look at this bug during the upcoming week.

**Comment by JohnnyQQQQ** [comment]
This will be fixed in 9.1.7 and 9.2.X
